Sources/MapboxMaps/Foundation/Extensions/UIWindow.swift

```diff
@@ -0,0 +1,9 @@
+import UIKit
+
+@available(iOS 13.0, *)
+extension UIWindow {
+    internal var parentScene: UIScene? {
+        // falling back to ``CPWindow.templateApplicationScene`` to also cover CarPlay
+        return windowScene ?? self.value(forKeyPath: "templateApplicationScene") as? UIScene
```

Comment:
This was a tricky one - we don't really know at the compile time if CarPlay is available. So it seems that resorting to runtime checks is the way to go. 

UIKit provides us only with `windowScene`, which is `nil` in case of CarPlay window. `CPWindow` has a separate scene property called `templateApplicationScene`. 
Thus the return value of this computed property will be `windowScene` in case of `UIWindow`, and `templateApplicationScene` in case of `CPWindow`. 

I'm open to suggestions on how to improve this :)